### ISSUE TITLE
vq-wav2vec: self-supervised learning of discrete speech representations

### ISSUE THREAD

**Issue by patrick-g-zhang** [patrick-g-zhang/papers-i-read#6]
# Abstract
- discrete representations through a wav2vec-style self-supervised context prediction task
- uses either a Gumbel-Softmax or online k-means clustering(VQ-VAE) to quantize the dense representations
# Method
![img](https://pbs.twimg.com/media/Ediho1gUMAAF1-0?format=png&name=large)
![img](https://pbs.twimg.com/media/EdipwDEU4AI4G9L?format=png&name=large)
## GUMBEL-SOFTMAX
-   Gumbel-Softmax enables selecting discrete codebook variables in a fully differentiable way and we use the straight-through estimator
- during the forward, pick argmax, during the backward using true gradient(可以保证倒数的传递)

## K-MEANS
- prone to mode collapse where only some of the codewords are actually used(this
problem has been mitigated by workarounds such as re-initializing codewords or applying additional
regularizers to the loss function )
- independently quantize partitions of z
- a new method which help prevent this isuse - grouping 

## BERT PRE-TRAINING ON QUANTIZED SPEECH
# Experiment
